lib/facility.rb

```diff
@@ -1,14 +1,77 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
+
+
+  def initialize(args)
+    @name = args[:name]
+    @address = args[:address]
+    @phone = args[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration') 
+      
+      if vehicle.antique?
+        cost = 25
+        plate_type = :antique
+      elsif vehicle.electric_vehicle?
+        cost = 200
+        plate_type = :ev
+      else
+        cost = 100
+        plate_type = :regular
+      end
+
+    vehicle.set_plate_type(plate_type)
+    vehicle.set_registration_date(Date.today)
+    @collected_fees += cost
+    @registered_vehicles << vehicle
+
+    end
+  end 
+
+  
+  def vehicle_registered?(vehicle)
+    @registered_vehicles.include?(vehicle)
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') && registrant.age >= 16 && registrant.permit == true
```

Comment:
Yes! This looks great and this is exactly how you would avoid nested if statements. Nice work with this! The only part I would fix, is I would call `registrant.permit?` at the end, since that returns a boolean, we want to use our method that ends in a `?`